lua/gen/init.lua

```diff
@@ -54,12 +54,28 @@ end
 M.command = 'ollama run $model $prompt'
 M.model = 'mistral:instruct'
 
+local function is_serving(pname)
+    local handle, ok = pcall(io.popen, 'ps aux | grep "' .. pname .. '" | grep -v grep')
+    if not ok or not handle then
+      return false
+    end
+    local result = handle:read("*a")
+    handle:close()
+    return result ~= ""
+end
```

Comment:
```suggestion
local function is_serving(pname)
    local handle, ok = pcall(io.popen, 'pgrep -q "' .. pname .. '"')
    if not ok or not handle then
      return false
    end
    handle:close()
    return true
end
```

This should be supported on both bsd (including macos) and gnu (including linux), and no need for all the greps

Reply:
Sorry, didn't mean to intervene, just a suggestion.